src/bsoncxx/document/view.cpp

```diff
@@ -124,7 +125,8 @@ view::const_iterator view::const_iterator::operator++(int) {
 }
 
 bool operator==(const view::const_iterator& lhs, const view::const_iterator& rhs) {
-    return (lhs._element.raw == rhs._element.raw && lhs._element.offset == rhs._element.offset);
+    return (lhs._element.raw() == rhs._element.raw() &&
```

Comment:
tie wants a reference, not a pointer. I'm guessing we actually want to compare the pointers here. Is there a way to do that with tie?